modules/install-couchbase-server/install-couchbase-server

```diff
@@ -65,29 +65,40 @@ function install_couchbase_on_ubuntu {
   local readonly version="$2"
   local readonly checksum="$3"
   local readonly checksum_type="$4"
+  local readonly ubuntu_version="$(lsb_release -r -s)"
+  local readonly dpkg_name="couchbase-server.deb"
 
-  log_info "Installing Couchbase $version ($edition edition) on Ubuntu"
+  log_info "Installing Couchbase $version ($edition edition) on Ubuntu $ubuntu_version"
 
-  local readonly filepath="couchbase-server-${edition}_${version}-ubuntu16.04_amd64.deb"
+  local readonly filepath="couchbase-server-${edition}_${version}-ubuntu${ubuntu_version}_amd64.deb"
   local readonly url="https://packages.couchbase.com/releases/$version/$filepath"
 
+  # NOTE: Couchbase only started publishing Ubuntu 18.04 deb packages for Enterprise edition version 6.0.2, which means
+  # that the vast majority of Couchbase server builds do not have official deb packages for 18.04. That said, the 16.04
+  # package has been verified to work, so we will fallback to using that if the 18.04 build is not available.
+  local readonly fallback_filepath="couchbase-server-${edition}_${version}-ubuntu16.04_amd64.deb"
+  local readonly fallback_url="https://packages.couchbase.com/releases/$version/$fallback_filepath"
+
   log_info "Installing Couchbase dependencies"
   sudo DEBIAN_FRONTEND=noninteractive apt-get update
   sudo apt-get install -y python-httplib2
 
   log_info "Downloading Couchbase from $url to $filepath"
-  curl --location --silent --fail --show-error -O "$url"
+  if ! curl --location --silent --fail --show-error -o "$dpkg_name" "$url"; then
```

Comment:
Should this explicitly check for a 404?

Reply:
That's probably a good idea! Do you know off the top of your head how to do that? If not, I can yak shave that when I get back to this.

Reply:
```
curl -s -o /dev/null -w "%{http_code}" http://www.example.org/
```

Reply:
Thanks for the tip. Added this in!